src/components/discover-sheet/ListsSection.js

```diff
@@ -0,0 +1,334 @@
+  return trendingAddresses;
+};
+
+const ListButton = styled(ButtonPressAnimation).attrs({
+  scaleTo: 0.96,
+})`
+  margin-right: 16px;
+  ${({ selected, theme: { colors } }) =>
+    selected
+      ? `
+        background-color: ${colors.alpha(colors.blueGreyDark, 0.06)};
+        border-radius: 12px;
+        height: 30px;
+        padding-horizontal: 8px;
+        padding-top: ${ios ? 6 : 4}px;
+      `
+      : `
+        padding-top: ${ios ? 6 : 4}px;
+      `}
+`;
+
+const ListName = styled(Text)`
+  margin-left: 3px;
+  margin-top: ${ios ? -4.5 : 0}px;
+`;
+
+// Update trending lists every 5 minutes
+const TRENDING_LIST_UPDATE_INTERVAL = 5 * 60 * 1000;
+
+const layouts = [
+  { length: 110, offset: 0 },
+  { length: 89, offset: 110 },
+  { length: 120, offset: 200 },
+  { length: 81, offset: 320 },
+  { length: 92, offset: 400 },
+];
+const getItemLayout = (_, index) => ({
+  index,
+  ...layouts[index],
+});
+
+export default function ListSection() {
+  const dispatch = useDispatch();
+  const { network, nativeCurrency } = useAccountSettings();
+  const { navigate } = useNavigation();
+  const {
+    favorites,
+    lists,
+    updateList,
+    ready,
+    selectedList,
+    setSelectedList,
+    clearList,
+  } = useUserLists();
+  const listRef = useRef(null);
+  const initialized = useRef(false);
+  const { allAssets } = useAccountAssets();
+  const { genericAssets } = useSelector(({ data: { genericAssets } }) => ({
+    genericAssets,
+  }));
+  const { colors } = useTheme();
+  const listData = useMemo(() => DefaultTokenLists[network], [network]);
+
+  const addressSocket = useSelector(
+    ({ explorer: { addressSocket } }) => addressSocket
+  );
+  useEffect(() => {
+    if (addressSocket !== null) {
+      Object.values(listData).forEach(({ tokens }) => {
+        dispatch(emitAssetRequest(tokens));
+        dispatch(emitChartsRequest(tokens));
+      });
+    }
+  }, [addressSocket, dispatch, listData]);
+
+  const trendingListHandler = useRef(null);
+
+  const updateTrendingList = useCallback(async () => {
+    const tokens = await fetchTrendingAddresses();
+    clearList('trending');
+
+    dispatch(emitAssetRequest(tokens));
+    dispatch(emitChartsRequest(tokens));
+    tokens.forEach(address => {
+      updateList(address, 'trending', true);
+    });
+
+    trendingListHandler.current = setTimeout(
+      () => updateTrendingList(),
+      TRENDING_LIST_UPDATE_INTERVAL
+    );
+  }, [clearList, dispatch, updateList]);
+
+  const handleSwitchList = useCallback(
+    (id, index) => {
+      LayoutAnimation.configureNext(
+        LayoutAnimation.create(200, 'easeInEaseOut', 'opacity')
+      );
+      setSelectedList(id);
+      listRef.current?.scrollToIndex({
+        animated: true,
+        index,
+        viewPosition: 0.5,
+      });
+    },
+    [setSelectedList]
+  );
+
+  useEffect(() => {
+    if (ready && !initialized.current) {
+      ready && updateTrendingList();
+      lists.forEach((list, index) => {
```

Comment:
Can we replace the forEach with lodash find here for an early exit once the selected list is found

Reply:
Can't we use a normal `find` from `Array.prototype`?

Reply:
it's a 4 element array but okay

Reply:
Actually I'm gonna use `findIndex` cause I need the index too.

Reply:
great! yeah lodash vs Array.prototype doesn't matter so much - findIndex sounds good!